integrations/node/workload-executor.js

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env node
+
+const MongoClient = require('../../node-mongodb-native').MongoClient;
+const fs = require('fs');
+const assert = require('assert');
+const omit = require('lodash.omit');
+
+let interrupted = false;
+process.on('SIGINT', () => (interrupted = true));
+
+const executors = {
+  find: (collection, args) =>
+    collection.find(args.filter || {}, omit(args, ['filter'])).toArray(),
+
+  insertOne: (collection, args) =>
+    collection.insertOne(args.document, { forceServerObjectId: true }),
+
+  updateOne: (collection, args) =>
+    collection.updateOne(args.filter, args.update)
+};
+
+async function main(uri, spec) {
+  const runOperations = makeRunOperations(spec);
+
+  const client = new MongoClient(uri);
+  await client.connect();
```

Comment:
just in terms of cleanliness, I think you want to wrap this entire `main` in a try/catch/finally, to ensure that the client is always closed:
```
async function main(..) {
  cons client = new MongoClient(...);

  try {
    await client.connect();
    ... other stuff ...
  } catch (err) { 
    ... handle error ...
  } finally {
    await client.close();
  }

  process.exit(0);
}
```